packages/aws-cdk/test/context-providers/generic.test.ts

```diff
@@ -27,6 +27,36 @@ test('errors are reported into the context value', async () => {
   expect(context.get('asdf').$providerError).toBe('Something went wrong');
 });
 
+test('lookup role ARN is resolved', async () => {
+  // GIVEN
+  contextproviders.registerContextProvider(TEST_PROVIDER, class {
+    public async getValue(args: {[key: string]: any}): Promise<any> {
+      if (args.lookupRoleArn != null) {
```

Comment:
This is not testing that the partition has been resolved to the right value. For all we know there could still be `${AWS::Partition}` in there.